<launch>
	<!-- Arguments -->
	<arg name="baud_rate"  		      default="0"     doc="baud rate value [0:(3,000,000), 1:(921,600), 2:(115,200), 3:(57,600)]"/>
	<arg name="run_mode"              default="2"     doc="version type [0(2D), 1(3D), 2(2D/3D)]"/>
	<arg name="frequency_channel"     default="0"     doc="frequency Ch. [0 to 15]"/>
	<arg name="duration_mode"         default="0"     doc="pulse mode [0 (Auto), 1(Fixed)]"/>
	<arg name="duration_value"        default="10000" doc="duration [0 to 10000]"/>
	<arg name="color_mode"    	      default="0" 	  doc="color mode [0 (HUE), 1 (RGB), 2 (GRAY)]"/>
	<arg name="data_type_3d"          default="0" 	  doc="3D data type [0 (DISTANCE), 1 (AMPLITUDE)]"/>
	<arg name="filter_mode"           default="0" 	  doc="New 3D Data Filtering [0 (None), 1 (Median Filter), 2 (Average Filter)]"/>
	<arg name="edge_filter_value"     default="0" 	  doc="Edge Filtering Treshhold"/>
	<arg name="enable_kalmanfilter"   default="true"  doc="Kalman Filtering [false (Raw Distance Data), true (Kalman Filter Applied Distance Data)]"/>
	<arg name="enable_clahe" 		  default="false" doc="Ampliutde 3D type [false (Raw Amplitude Data), true (CLAHE Applied Amplitude Data)]"/>
	<arg name="clahe_cliplimit" 	  default="40"    doc="ClipLimit from 0 to 255 / Default : 40"/>
	<arg name="clahe_tiles_grid_size" default="8" 	  doc="Tiles Grid Size from 1 to 15 / Default : 8"/>

	<node name="cyglidar_publisher" pkg="cyglidar_d2_ros1" type="cyglidar_publisher" output="screen">
		<param name="port_number"       	value="/dev/ttyUSB0"/>
		<param name="baud_rate"         	value="$(arg baud_rate)"/>
		<param name="frame_id"          	value="laser_frame"/>
		<param name="fixed_frame"       	value="/map"/>
		<param name="run_mode"          	value="$(arg run_mode)"/>
		<param name="frequency_channel" 	value="$(arg frequency_channel)"/>
		<param name="duration_mode"     	value="$(arg duration_mode)"/>
		<param name="duration_value"    	value="$(arg duration_value)"/>
		<param name="color_mode" 			value="$(arg color_mode)"/>
		<param name="data_type_3d" 			value="$(arg data_type_3d)"/>
		<param name="filter_mode" 			value="$(arg filter_mode)"/>
		<param name="edge_filter_value" 	value="$(arg edge_filter_value)"/>
		<param name="enable_kalmanfilter"   value="$(arg enable_kalmanfilter)"/>
		<param name="enable_clahe" 		    value="$(arg enable_clahe)"/>
		<param name="clahe_cliplimit"       value="$(arg clahe_cliplimit)"/>
		<param name="clahe_tiles_grid_size" value="$(arg clahe_tiles_grid_size)"/>
	</node>

	<node name="to_laserframe" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 map laser_frame 10"/>
</launch>
